src/main.js

```diff
@@ -37,6 +42,33 @@ async function run() {
         body += `\n:no_entry: ${errorMessage}`;
       }
 
+      const updateGitHubComment = commentId =>
+      octokit.issues.updateComment({
+        repo: github.context.repo.repo,
+        owner: github.context.repo.owner,
+        comment_id: commentId,
+        body,
+      })
+
+      if (updateComment == "true") {
+        const issueComments = await octokit.issues.listComments({
+          repo: github.context.repo.repo,
+          owner: github.context.repo.owner,
+          issue_number: github.context.payload.pull_request.number,
+        })
+
+        const existingComment = issueComments.data.find(comment =>
+          comment.body.includes(commentIdentifier(process.env.GITHUB_WORKFLOW)),
+        )
+
+        if (existingComment) {
+          console.log('Update Comment ID: ' + existingComment.id);
+          await updateGitHubComment(existingComment.id);
+          return
```

Comment:
Since we are returning here, the failing handling in line 80 does not get executed. I am not sure why that was added in the first place. May be @zgosalvez  can answer here. But if that is needed, we should not return here.

Reply:
@balaji-nordic I've proposed improvements in https://github.com/zgosalvez/github-actions-report-lcov/pull/42. The failure handling in line 80 is about checking if minimum coverage is satisfied. Comment should be created/updated either way.